lib/src/realm_object.dart

```diff
@@ -178,7 +178,7 @@ class RealmCoreAccessor implements RealmAccessor {
   void set(RealmObject object, String name, Object? value, {bool isDefault = false, bool update = false}) {
     final propertyMeta = metadata[name];
     try {
-      if (value is RealmList<Object>) {
+      if (value is RealmList<Object?>) {
```

Comment:
if `value is RealmList<Object?>` then value is not `RealmList<Object>`. Are we sure we are not breaking the old behavior? Should this be 
```dart
if (value is RealmList<Object?> || value is RealmList<Object>)
```

Reply:
```dart
value is RealmList<Object>
```
implies
```dart
value is RealmList<Object?>
```
https://dartpad.dev/?id=d9aecd56b5a115b6a5672f0fbefbd49f

Reply:
I don't understand the answer. We are changing a check from `if (value is RealmList<Object>)` to `if (value is RealmList<Object?>)`. Which now does not capture  `value is RealmList<Object>` . Isn't this a issue?

Reply:
The way I understand the answer, if I have `final foo = RealmList<Object>()`, then `foo is RealmList<Object?>` will return `true`. So the new check is a looser one - it will return `true` for both `RealmList<Object>` and `RealmList<Object?>`. Looking at the dartpad link, you can see that the compiler correctly detects that and gives you a hint saying `Unnecessary type check; the result is always 'true'.`.

Reply:
Correct

Reply:
I have added an extra test case that demonstrates covariance of realm lists.

@blagoev What may confuse you is that in dart generics are covariant by default.

Reply:
https://dart.dev/guides/language/type-system#generic-type-assignment

Reply:
Thanks. Indeed. I was testing the other way around. Then there is no issue